UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using AddressableReferences;
+using Mirror;
+using UnityEngine;
+
+[RequireComponent(typeof(Pickupable))]
+public class StunBatonV2 : NetworkBehaviour, ICheckedInteractable<HandActivate>, IServerSpawn
+{
+	public SpriteHandler spriteHandler;
+
+    public MeleeStun meleeStun;
+	
+	// Sound played when turning this baton on/off.
+	public AddressableAudioSource ToggleSound;
+
+	///Both used as states for the baton and for the sub-catalogue in the sprite handler.
+	private enum BatonState
+	{
+		Off,
+		On,
+		NoCell
+	}
+
+	private BatonState batonState;
+
+    private void Awake()
+    {
+        meleeStun = GetComponent<MeleeStun>();
+    }
+
+	// Calls TurnOff() when baton is spawned, see below.
+	public void OnSpawnServer(SpawnInfo info)
+	{
+		TurnOff();
+	}
+
+	// Enables melee stun component and sets baton state to on.
```

Comment:
no need to over explain the code. This code is perfectly legible, I read a method called "TurnOn" inside the "StunBaton" class, I can already imagine what it does. If I need more details, I read the method body.

Reply:
Fixed in fe4b08d. Removed a few unnecessary comments.

Reply:
I'd remove all comments, tbh. That's fine tho, I'm merging.